python/cudf/cudf/utils/dtypes.py

```diff
@@ -377,6 +386,47 @@ def is_dtype_obj_numeric(
         return is_non_decimal
 
 
+try:
+    pa_decimal32type = pa.Decimal32Type
+except AttributeError:
+    pa_decimal32type = None
+
+try:
+    pa_decimal64type = pa.Decimal64Type
+except AttributeError:
+    pa_decimal64type = None
+
+
+def pyarrow_dtype_to_cudf_dtype(dtype: pa.DataType) -> DtypeObj:
+    """Given a pyarrow dtype, converts it into the equivalent cudf pandas
+    dtype.
+    """
+
+    pyarrow_dtype = dtype.pyarrow_dtype
+    if isinstance(pyarrow_dtype, pa.Decimal128Type):
+        return cudf.Decimal128Dtype.from_arrow(pyarrow_dtype)
+    elif pa_decimal64type is not None and isinstance(
+        pyarrow_dtype, pa_decimal64type
+    ):
+        return cudf.Decimal64Dtype.from_arrow(pyarrow_dtype)
+    elif pa_decimal32type is not None and isinstance(
+        pyarrow_dtype, pa_decimal32type
+    ):
+        return cudf.Decimal32Dtype.from_arrow(pyarrow_dtype)
+    elif isinstance(pyarrow_dtype, pa.ListType):
+        return cudf.ListDtype.from_arrow(pyarrow_dtype)
+    elif isinstance(pyarrow_dtype, pa.StructType):
+        return cudf.StructDtype.from_arrow(pyarrow_dtype)
+    elif str(pyarrow_dtype) == "large_string":
+        return CUDF_STRING_DTYPE
+    elif isinstance(pyarrow_dtype, pa.DataType):
+        return pyarrow_dtype.to_pandas_dtype()
+    elif pyarrow_dtype is pa.date32():
+        return TypeError("Unsupported type")
```

Comment:
```suggestion
```

Reply:
Did you mean to suggest something else or delete this?

Reply:
I think this is unneeded since the last condition should cover this case (plus I think you'd want to `raise` instead of `return` this exception)

Reply:
The intention of the if statement is to raise here an not go through `to_pandas_type` since that returns an incompatible dtype for cudf.